templates/_macros/hero.jinja

```diff
@@ -0,0 +1,115 @@
+{% from "_macros/cta/cta-block.jinja" import cta_block %}
+{% from "_macros/image-container.jinja" import image_container %}
+
+# Params
+# title_text: Hero title text (required)
+# subtitle_text: Hero subtitle text
+# is_full_width_on_tablet: whether image is in its own row on tablet.
+#   If false, image will be in its own row (stacked below header/description)
+# image_src: image source URL (required)
+# image_alt: image alt text
+# image_aspect_ratio: image aspect ratio (16/9, 3/2, 2/3, cinematic, square)
+# image_full_cover: whether the image should span the full width of the bottom of the hero section
+# image_hidden_on_devices: list of devices where the image should be hidden, e.g. ["desktop", "tablet"] to only show on small screens
+# cta_primary_url: primary CTA button URL
+# cta_secondary_url: secondary CTA button URL
+# Slots
+# description: slot for additional paragraph-style content below the title and subtitle
+
+{% macro hero(
+  title_text="",
```

Comment:
Most of the content will have to be passed to slots as it would require HTML.

For images also, we probably just need to allow a slot for image, and let devs use image container.
Especially that they will want to use image template module to serve image, not pass src to img directly.

As far as I understand the only params we want is flags that define the layout/variant of the component. Any kind of content should be HTML passed to slots.

Reply:
Image content passed in by slot now. 

Title passing to be discussed as part of [this conversation](https://github.com/canonical/vanilla-framework/pull/5189#discussion_r1683908905)